apps/antalmanac/src/stores/Schedules.ts

```diff
@@ -175,14 +175,18 @@ function getColorForNewSection(newSection: ScheduleCourse, sectionsInSchedule: S
                 Math.abs(parseInt(b.section.sectionCode) - parseInt(newSection.section.sectionCode))
         );
 
-    const usedColors = new Set(sectionsInSchedule.map((course) => course.section.color));
-
-    if (existingSections.length > 0) {
-        const originalColor = existingSections[0].section.color;
+    //New array of courses that share the same sectionType & courseTitle
+    const existingSectionsType = existingSections.filter((course) => course.section.sectionType === newSection.section.sectionType)
 
-        // Generate a slightly different color that does not conflict with any other section in the schedule
-        return generateCloseColor(originalColor, usedColors);
-    }
+    const usedColors = new Set(sectionsInSchedule.map((course) => course.section.color));
+    
+    //If the same sectionType exists, return that color
+    if(existingSectionsType.length > 0) 
+        return (existingSectionsType[0].section.color);
+        
+    //If the same courseTitle exists, but not the same sectionType, return a hue
```

Comment:
```suggestion
    //If the same courseTitle exists, but not the same sectionType, return a close color
```

It's the same hue and saturation with a different lightness (currently). I prefer not to say that the function returns a hue of the original color, which would imply that it's a different hue.